src/dcap_provider.cpp

```diff
@@ -203,34 +205,46 @@ static inline quote3_error_t fill_qpl_string_buffer(
 //
 // Determine time cache should invalidate for given collateral
 //
-bool get_cache_expiration_time(CollateralTypes collateral_type, time_t &expiration_time)
+bool get_cache_expiration_time(const string &cache_control, const string &url, time_t &expiration_time)
 {
     time_t max_age = 0;
     tm* max_age_s = localtime(&max_age);
-
-    switch(collateral_type)
+    string match = "max-age=";
+    size_t index = cache_control.find(match);
+    int cache_time_seconds = 0;
+    if (index != string::npos)
     {
-        case CollateralTypes::TcbInfo:
-        case CollateralTypes::QeIdentity:
-        case CollateralTypes::QveIdentity:
+        try 
         {
-            max_age_s->tm_hour += 12;
-            break;
+            cache_time_seconds = stoi(cache_control.substr(index + match.length()));
         }
-        case CollateralTypes::PckCert:
-        case CollateralTypes::PckCrl:
-        case CollateralTypes::PckRootCrl:
+        catch (std::invalid_argument e)
         {
-            max_age_s->tm_mday += 1;
-            break;
+            log(SGX_QL_LOG_ERROR,
+                "Invalid argument thrown when parsing cache-control. Header text: '%s' Error: '%s'",
+                cache_control.c_str(),
+                e.what());
+            cache_time_seconds = 0;

```

Comment:
>cache_time_seconds = 0; [](start = 12, length = 23)

cache_time_seconds is local. Setting it before returning has no meaning.

Reply:
Thanks! fixed